docs/utilities/recipes.md

```diff
@@ -0,0 +1,135 @@
+```
+
+Unlocking Recipes within the Recipe Book
+----------------------------------------
+When you first obtain an ingredient to a vanilla recipe it will automatically unlock the recipe in the
+recipe book. To achieve the same affect, you have to use the [`Advancement`][Advancements] system and create
+a new `Advancement` for each of your ingredients.
+
+!!! note
+
+	The advancement has to exist. This doesn't mean it has to be visible in the advancement tree.
+
+Item Objects
+------------
+
+Type
+----
+TODO
+	"type": "forge:oreshaped",
+	
+Groups
+------
+
+Shaped and shapeless crafting
+-----------------------------
+Within this section we will take a closer look on the differences between defining a shaped and a shapeless crafting recipe. Please read both sections carefully as I won't explain the same keywords twice when they can be used within both recipe types!
+
+### Shaped crafting
+Shaped recipes require the `pattern` and `key` keywords. A pattern defines the slot an item must appear in using placeholder characters. You can choose whatever character you want to be a placeholder for an item. Keys on the other hand define what items are to be used instead of the placeholders. A key is defined by a placeholder character and the item. 
+Additional the type `forge:ore_dict` may be added. This defines the item beeing part of the [`OreDictionary`][OreDictionary] and can for example be used when it doesn't matter which copper ore is used to produce a copper ingot. In this case the `ore` tag has to be used instead of the `item` tag to define the item.
+The `data` tag is a optional and used to define the metadata of a block or item.
+
+### Shapeless crafting
+A shapeless recipe doesn't make use of the `pattern` and `key` keywords. 
+
+To define a shapeless recipe, you have to use the `ingredients` list. It defines which items have to be used for the crafting process and can also make use of the additional type `forge:ore_dict` and it's functionality as described above. It is even possible to define multiply instances of the same item which means multiply of these items have to be in place for the crafting recipe to take place. 
+
+!!! note
+
+	While there is no limit on how many ingredients your recipe requires the vanilla crafting table does only allow 9 items to be placed for each crafting recipe.
+
+The following example shows how an ingredient list looks like within JSON.
+
+```json
+    "ingredients": [
+        {
+            "type": "forge:ore_dict",
+            "ore": "minecraft:diamond"
+        },
+        {
+            "item": "minecraft:nether_star"
+        }
+    ],
+```
+
+### Special behaviour of the `data` tag
+It is strongly adviced to use the `data` tag to define the metadata of your items and blocks. When it is not used within the ingredients or keys, it will mean any metadata of this item will be accepted, for example: Not defining the data of a sword means even a half broken sword will be accepted for the crafting recipe!
```

Comment:
Important to mention that any item which calls `setHasSubtypes(true)` will require the data field.